In [126]:
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt
plt.style.use(style='ggplot')
#plt.rcParams['figure.figsize'] = (30, 20)
%matplotlib inline

In [127]:
# Loading the data into the Pandas dataframe
mlsdata = pd.read_csv("unique_mls_db_GIT.csv")
#Let's take look into first 10 rows of data
mlsdata.head(10)

mlsno  stno        stname         city             area  soldprice  \
0  E3853125    46       Hinsley         Ajax   Northeast Ajax     595000   
1  E3833751   508       Norfolk    Pickering         Amberlea     677000   
2  N3788507    36     Innisvale      Markham          Cornell     725000   
3  E3855735    14       Shudell  Toronto E01      Blake-Jones     755000   
4  W3854388  2500       Hargood  Mississauga     Streetsville    1050000   
5  W3830580   386  Landsborough       Milton            Scott     670000   
6  W3698326   184        Dianne     Oakville         Eastlake    4350000   
7  W3843110  2382     Baronwood     Oakville  West Oak Trails     754000   
8  W3833449   364    Strathcona   Burlington       Shoreacres     780000   
9  E3855200   786        Wesley       Oshawa          Donevan     311000   

     solddate              type        style  bdrm  wshrm   latitude  \
0  2017-06-30     Semi-Detached     2-Storey     3      3  43.891599   
1  2017-06-30          Detached     2-Storey     3      3  43.827530   
2  2017-06-30     Semi-Detached     2-Storey     4      3  43.900148   
3  2017-06-30     Semi-Detached     2-Storey     2      2  43.675379   
4  2017-06-30          Detached     2-Storey     4      3  43.576264   
5  2017-06-30  Att/Row/Twnhouse     2-Storey     3      3  43.491307   
6  2017-06-30          Detached     2-Storey     4      6  43.460869   
7  2017-06-30  Att/Row/Twnhouse     2-Storey     3      3  43.432442   
8  2017-06-30          Detached  Backsplit 3     3      2  43.352552   
9  2017-06-29          Detached     Bungalow     2      2  43.887303   

   longitude house_condo aptno                        address  Unnamed: 16  \
0 -79.028612       house   NaN          46 Hinsley Cres, Ajax          NaN   
1 -79.123617       house   NaN      508 Norfolk Sq, Pickering          NaN   
2 -79.238976       house   NaN       36 Innisvale Dr, Markham          NaN   
3 -79.337634       house   NaN        14 Shudell Ave, Toronto          NaN   
4 -79.729552       house   NaN   2500 Hargood Pl, Mississauga          NaN   
5 -79.885066       house   NaN   386 Landsborough Ave, Milton          NaN   
6 -79.656954       house   NaN       184 Dianne Ave, Oakville          NaN   
7 -79.769478       house   NaN    2382 Baronwood Dr, Oakville          NaN   
8 -79.764853       house   NaN  364 Strathcona Dr, Burlington          NaN   
9 -78.807093       house   NaN          786 Wesley Dr, Oshawa          NaN   

   Unnamed: 17  Unnamed: 18  
0          NaN          NaN  
1          NaN          NaN  
2          NaN          NaN  
3          NaN          NaN  
4          NaN          NaN  
5          NaN          NaN  
6          NaN          NaN  
7          NaN          NaN  
8          NaN          NaN  
9          NaN          NaN

In [128]:
# Getting description of the data attributes:
mlsdata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195524 entries, 0 to 195523
Data columns (total 19 columns):
mlsno          195524 non-null object
stno           195523 non-null object
stname         195524 non-null object
city           195524 non-null object
area           195524 non-null object
soldprice      195524 non-null int64
solddate       195524 non-null object
type           195523 non-null object
style          195179 non-null object
bdrm           195524 non-null int64
wshrm          195524 non-null int64
latitude       195524 non-null float64
longitude      195524 non-null float64
house_condo    195524 non-null object
aptno          68026 non-null object
address        195524 non-null object
Unnamed: 16    0 non-null float64
Unnamed: 17    0 non-null float64
Unnamed: 18    0 non-null float64
dtypes: float64(5), int64(3), object(11)
memory usage: 28.3+ MB


As we can see from the data, the attriute "type" have "Condo", "semi detached", "detached" etc. We can use this title for a feature engineering. From the descriptive statistics table above, we can see that the attribute contains 195523 unique values. The attribute House_condo in data has defined houses in two types as house and condo.
There are multiple ways of approaching this problem. Let's implement the approach of using house types to group them into four common groups. We will creat a new feature name house_types in dataframe and we can drop the feature type in future analysis.

In [141]:
items_counts = mlsdata['type'].value_counts()
print(items_counts)

Detached              84506
Condo Apt             55410
Semi-Detached         19915
Att/Row/Twnhouse      14976
Condo Townhouse       14566
Link                   2500
Comm Element Condo     1616
Vacant Land             371
Duplex                  300
Co-Op Apt               285
Co-Ownership Apt        146
Triplex                 131
Parking Space           109
Store W/Apt/Offc         95
Det W/Com Elements       95
Multiplex                95
Det Condo                69
Semi-Det Condo           62
Leasehold Condo          45
Other                    43
Fourplex                 38
Rural Resid              36
Farm                     30
Mobile/Trailer           22
Cottage                  20
Locker                   15
Condo Ap                  9
Vacant Lan                5
Semi-Detac                4
Condo To                  2
Leasehol                  2
Parking                   1
Semi-Det                  1
Condo Apartment           1
Rural Resi                1
Mobile/Tra          

In [142]:
def generate_new_types(list_of_types):
    list_of_new_types = []
    for x in list_of_types:
        x = str(x)
        if any(x.lstrip() in s for s in new_type['Condo']):
            list_of_new_types.append("Condo")
        elif any(x.lstrip() in s for s in new_type['Semidetached']):
            list_of_new_types.append("Semidetached")
        elif any(x.lstrip() in s for s in new_type['Detached']):
            list_of_new_types.append("Detached")
        elif any(x.lstrip() in s for s in new_type['Townhouse']):
            list_of_new_types.append("Townhouse")
        else:                                                   #elif any(x.lstrip() in s for s in new_type['Other']):
            list_of_new_types.append("Other")
    return list_of_new_types
new_list_types = generate_new_types(mlsdata['type'])
mlsdata['house_types'] = new_list_types

In [143]:
mlsdata.head(10)

mlsno  stno        stname         city             area  soldprice  \
0  E3853125    46       Hinsley         Ajax   Northeast Ajax     595000   
1  E3833751   508       Norfolk    Pickering         Amberlea     677000   
2  N3788507    36     Innisvale      Markham          Cornell     725000   
3  E3855735    14       Shudell  Toronto E01      Blake-Jones     755000   
4  W3854388  2500       Hargood  Mississauga     Streetsville    1050000   
5  W3830580   386  Landsborough       Milton            Scott     670000   
6  W3698326   184        Dianne     Oakville         Eastlake    4350000   
7  W3843110  2382     Baronwood     Oakville  West Oak Trails     754000   
8  W3833449   364    Strathcona   Burlington       Shoreacres     780000   
9  E3855200   786        Wesley       Oshawa          Donevan     311000   

     solddate              type        style  bdrm  wshrm   latitude  \
0  2017-06-30     Semi-Detached     2-Storey     3      3  43.891599   
1  2017-06-30          Detached     2-Storey     3      3  43.827530   
2  2017-06-30     Semi-Detached     2-Storey     4      3  43.900148   
3  2017-06-30     Semi-Detached     2-Storey     2      2  43.675379   
4  2017-06-30          Detached     2-Storey     4      3  43.576264   
5  2017-06-30  Att/Row/Twnhouse     2-Storey     3      3  43.491307   
6  2017-06-30          Detached     2-Storey     4      6  43.460869   
7  2017-06-30  Att/Row/Twnhouse     2-Storey     3      3  43.432442   
8  2017-06-30          Detached  Backsplit 3     3      2  43.352552   
9  2017-06-29          Detached     Bungalow     2      2  43.887303   

   longitude house_condo aptno                        address  Unnamed: 16  \
0 -79.028612       house   NaN          46 Hinsley Cres, Ajax          NaN   
1 -79.123617       house   NaN      508 Norfolk Sq, Pickering          NaN   
2 -79.238976       house   NaN       36 Innisvale Dr, Markham          NaN   
3 -79.337634       house   NaN        14 Shudell Ave, Toronto          NaN   
4 -79.729552       house   NaN   2500 Hargood Pl, Mississauga          NaN   
5 -79.885066       house   NaN   386 Landsborough Ave, Milton          NaN   
6 -79.656954       house   NaN       184 Dianne Ave, Oakville          NaN   
7 -79.769478       house   NaN    2382 Baronwood Dr, Oakville          NaN   
8 -79.764853       house   NaN  364 Strathcona Dr, Burlington          NaN   
9 -78.807093       house   NaN          786 Wesley Dr, Oshawa          NaN   

   Unnamed: 17  Unnamed: 18   house_types  
0          NaN          NaN  Semidetached  
1          NaN          NaN  Semidetached  
2          NaN          NaN  Semidetached  
3          NaN          NaN  Semidetached  
4          NaN          NaN  Semidetached  
5          NaN          NaN     Townhouse  
6          NaN          NaN  Semidetached  
7          NaN          NaN     Townhouse  
8          NaN          NaN  Semidetached  
9          NaN          NaN  Semidetached

In [145]:
#dropping the data rows contain "Other" house type and saving the dataframe into new data. Becasue most of the types are vacant land
# and parking lot etc. which don't provide any significant information for the hour type.
house_df = mlsdata[mlsdata.house_types != 'Other']

In [146]:
mlsdata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195524 entries, 0 to 195523
Data columns (total 20 columns):
mlsno          195524 non-null object
stno           195523 non-null object
stname         195524 non-null object
city           195524 non-null object
area           195524 non-null object
soldprice      195524 non-null int64
solddate       195524 non-null object
type           195523 non-null object
style          195179 non-null object
bdrm           195524 non-null int64
wshrm          195524 non-null int64
latitude       195524 non-null float64
longitude      195524 non-null float64
house_condo    195524 non-null object
aptno          68026 non-null object
address        195524 non-null object
Unnamed: 16    0 non-null float64
Unnamed: 17    0 non-null float64
Unnamed: 18    0 non-null float64
house_types    195524 non-null object
dtypes: float64(5), int64(3), object(12)
memory usage: 29.8+ MB
